nucypher/blockchain/eth/agents.py

```diff
@@ -319,15 +327,36 @@ def get_all_active_stakers(self, periods: int, pagination_size: Optional[int] =
             start_index: int = 0
             n_tokens: int = 0
             stakers: Dict[int, int] = dict()
-            active_stakers: Tuple[NuNits, List[List[int]]]
+            attempts = 0
             while start_index < num_stakers:
-                active_stakers = self.contract.functions.getActiveStakers(periods, start_index, pagination_size).call()
-                temp_locked_tokens, temp_stakers = active_stakers
-                # temp_stakers is a list of length-2 lists (address -> locked tokens)
-                temp_stakers_map = {address: locked_tokens for address, locked_tokens in temp_stakers}
-                n_tokens = n_tokens + temp_locked_tokens
-                stakers.update(temp_stakers_map)
-                start_index += pagination_size
+                try:
+                    attempts += 1
+                    active_stakers = self.contract.functions.getActiveStakers(periods,
+                                                                              start_index,
+                                                                              pagination_size).call()
+                except Exception as e:
+                    if 'timeout = 5s' not in str(e):
```

Comment:
This seems a bit fragile, since it depends on the exception containing this string exactly.  Not an RFC, just commentary.

Reply:
I hear ya - feels like we can do one of the following:
- simply check for the word "timeout" instead of including the length (it is currently a geth setting that can't be set)
OR
- don't perform any specific check on the text, and assume that any exception is caused by the timeout (i.e. remove the "if" check')
OR
...?